libs/lod/src/lib/services/prefix.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@angular/core';
+
+import { LodContext } from '../models';
+import { isCurie } from '../utils';
+
+@Injectable({ providedIn: 'root' })
+export class PrefixService {
+  private context: LodContext = {
+    prefix: {},
+  };
+
+  setContext(context: LodContext): void {
+    this.context = context;
+  }
+
+  compactUri(uri: string, context: LodContext = this.context): string {
+    if (isCurie(uri)) return uri;
+    for (const [prefix, value] of Object.entries(context.prefix ?? {}).sort(([, v1], [, v2]) => v2.length - v1.length)) {
```

Comment:
can we have 1 private method for the for loop being used in the `compactUri` with sort and `expandUri` or do you think it's a bit of an overkill here? 🤔

Reply:
and also, can we `if (context.base && uri.startsWith(context.base)) return uri.replace(context.base, '');`, after line 17? 🤔

Reply:
@srikanthv02 

> can we have 1 private method for the for loop being used in the compactUri with sort and expandUri or do you think it's a bit of an overkill here? 🤔
Seeing the length of the method, I believe that would be overkill 😅 

> and also, can we if (context.base && uri.startsWith(context.base)) return uri.replace(context.base, '');, after line 17? 🤔
That would makes sense indeed, will do 👍